.github/scripts/determine-initial-guest-os-versions.py

```diff
@@ -0,0 +1,193 @@
+import json
+import sys
+from enum import Enum
+from typing import Any, Dict, List, Optional, TypedDict, cast
+from urllib.request import urlopen
+
+ROLLOUT_DASHBOARD_ENDPOINT='https://rollout-dashboard.ch1-rel1.dfinity.network/api/v1/rollouts'
+PUBLIC_DASHBOARD_ENDPOINT='https://ic-api.internetcomputer.org/api/v3/subnets?format=json'
+
+# Key definitions
+EXECUTED_TIMESTAMP_SECONDS = 'executed_timestamp_seconds'
+REPLICA_VERSIONS = 'replica_versions'
+REPLICA_VERSION_ID = 'replica_version_id'
+SUBNETS = 'subnets'
+
+# Minimal subset of API structure needed for rollout dashboard.
+# Always keep me in sync with https://github.com/dfinity/dre-airflow/blob/main/rollout-dashboard/server/src/types.rs
+# We do not expect to change the API in ways that break code.
+class SubnetRolloutState(Enum):
+    error = "error"
+    predecessor_failed = "predecessor_failed"
+    pending = "pending"
+    waiting = "waiting"
+    proposing = "proposing"
+    waiting_for_election = "waiting_for_election"
+    waiting_for_adoption = "waiting_for_adoption"
+    waiting_for_alerts_gone = "waiting_for_alerts_gone"
+    complete = "complete"
+    unknown = "unknown"
+
+class Subnet(TypedDict):
+    subnet_id: str
+    git_revision: str
+    state: SubnetRolloutState
+
+class Batch(TypedDict):
+    subnets: List[Subnet]
+    # The following three are dates but they are ISO UTF Z,
+    # so they sort alphabetically.  Heh.
+    planned_start_time: str
+    actual_start_time: Optional[str]
+    end_time: Optional[str]
+
+class RolloutState(Enum):
+    complete = "complete"
+    failed = "failed"
+    preparing = "preparing"
+    upgrading_subnets = "upgrading_subnets"
+    upgrading_unassigned_nodes = "upgrading_unassigned_nodes"
+    waiting = "waiting"
+    problem = "problem"
+
+class Rollout(TypedDict):
+    name: str
+    state: RolloutState
+    batches: Dict[str, Batch]
+
+# Minimal subset of API structure needed for public dashboard.
+# Swagger for the public dashboard API: https://ic-api.internetcomputer.org/api/v3/swagger .
+class PDReplicaVersion(TypedDict):
+    executed_timestamp_seconds: Optional[int]
+    proposal_id: str  # really an int
+    replica_version_id: str
+
+class PDSubnet(TypedDict):
+    replica_versions: List[PDReplicaVersion]
+    subnet_id: str
+
+class PDSubnetsResponse(TypedDict):
+    subnets: List[PDSubnet]
+
+def eprint(*args, **kwargs):
+    print(*args, file=sys.stderr, **kwargs)
+
+def eprint_fmt(str, *args):
+    return # remove me to get some real action
+    print((str % args) if args else str, file=sys.stderr)
```

Comment:
Why don't we just print all of that? I don't see why would we not want it to be as verbose as possible.

Reply:
It's thousands of lines.  Super noisy.